### DISCUSSION TITLE
juicefs disconect from Storj storage server

### DISCUSSION THREAD

**Discussion by contenidosonline** [juicedata/juicefs#1755] [General]
I'm setting juicefs with storj on ubuntu 20.04.
 Community version

**What happened**:

Everything seems to work fine, but after an hour or two juicefs disconnects and the directory where I set up the file system becomes inaccessible.

**ls /backup-server
ls: cannot access '/backup-server': The other end of the connection is not connected**
and the file system is seen as unmounted.

If I mount it again, it works again for a few hours.

juicefs --version
juicefs version 1.0.0-beta2+2022-03-04T03:00:41Z.9e26080
using sqlite3 backend
it's a vps on Oracle Cloud
it's arm64 version

Linux rdp 5.13.0-1025-oracle #30~20.04.1-Ubuntu SMP Fri Apr 1 06:09:46 UTC 2022 aarch64 aarch64 aarch64 GNU/Linux
[dmesg.log.gz](https://github.com/juicedata/juicefs/files/8451207/dmesg.log.gz)
[juicefs-log.txt.gz](https://github.com/juicedata/juicefs/files/8451208/juicefs-log.txt.gz)

**Comment by SandyXSD** [comment]
The reason is that result of Storj ListObjects is not sorted, while S3 is. See: https://github.com/storj/roadmap/issues/20

You'd better disable auto-backup of JuiceFS by adding `--backup-meta 0` option when mounting. Moreover, don't use juicefs commands like destroy, gc and fsck since they require list sorted as well.

**Comment by solracsf** [comment]
See also https://github.com/juicedata/juicefs/issues/1655

**Comment by amwolff** [comment]
One note on this is that if you use the self-hosted S3 gateway (https://github.com/storj/gateway-st) to communicate with the Storj network, you can force results to be sorted with `--s3.fully-compatible-listing` (more details [here](https://github.com/storj/gateway-st/blob/main/docs/s3-compatibility.md#listobjects)).
